### DISCUSSION TITLE
Uninvited guest 'Csla.Blazor.WebAssembly.dll' is showing up to my Blazor Server Side party

### DISCUSSION THREAD

**Discussion by swegele** [MarimerLLC/csla#2767] [Questions] [answered]
1.  Create a new Blazor SS project in VS 2022 (no auth)
2.  Reference these two nuget packages (Csla.Blazor and Csla.AspNetCore)  
    choose latest version '6.0.0-R22020902'
3.  Build the solution
Notice the "Csla.Blazor.WebAssembly.dll" in the bin directory.  Why is it there?

That WebAssembly.dll being there by default is causing Csla to use the `Csla.Blazor.WebAssembly.ApplicationContextManager` 

If I delete that dll, then Csla properly uses the `Csla.AspNetCore.ApplicationContextManager`.  Every code-change/build puts that dll back again.

**Comment by swegele** [comment]
I see that the Package Csla.Blazor (6.0.0-R22020902) has the following in it's "Compile Time Assemblies":
- Csla.Blazor.dll
- Csla.Blazor.WebAssembly.dll

Tail wagging the dog?

Also, when I delete the Csla.Blazor.WebAssembly.dll so that Csla properly tries to use the Csla.AspNetCore.ApplicationContextManager, I get an error with service registration:
`InvalidOperationException: Unable to resolve service for type 'Csla.AspNetCore.Blazor.ActiveCircuitState' while attempting to activate 'Csla.AspNetCore.ApplicationContextManager'.`

I have to manually add this to my program.cs to get past that error:
`builder.Services.AddScoped<Csla.AspNetCore.Blazor.ActiveCircuitState>();`

**Reply by rockfordlhotka** [reply]
I just looked at the code, and this type is registered, along with the circuit handler type. You need to call the `AddAspNetCore` method within the `AddCsla` configuration though.

https://github.com/MarimerLLC/csla/blob/abc7d5b51d26d9e54e5a9923c5d369fd9e7d050e/Source/Csla.AspNetCore/ConfigurationExtensions.cs#L39

**Comment by rockfordlhotka** [comment, marked as answer]
You are right - the nuspec file has (had) a bug that included the .WebAssembly files where they aren't wanted.
